File: BMI_Tipper/app/src/main/java/com/example/tipper/BmiHistoryActivity.java
package com.example.tipper;

import android.graphics.Color;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.jjoe64.graphview.GraphView;
import com.jjoe64.graphview.GridLabelRenderer;
import com.jjoe64.graphview.LegendRenderer;
import com.jjoe64.graphview.helper.StaticLabelsFormatter;
import com.jjoe64.graphview.series.DataPoint;
import com.jjoe64.graphview.series.LineGraphSeries;

public class BmiHistoryActivity extends AppCompatActivity {
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bmi_history);
        GraphView graph = findViewById(R.id.graph);
        StaticLabelsFormatter staticLabelsFormatter = new StaticLabelsFormatter(graph);
        LineGraphSeries<DataPoint> series = new LineGraphSeries<>(new DataPoint[]{
                new DataPoint(0, 40),
                new DataPoint(1, 30),
                new DataPoint(2, 50),
                new DataPoint(3, 35),
                new DataPoint(4, 45),
        });

        graph.addSeries(series);
        staticLabelsFormatter.setHorizontalLabels(new String[]{
                "January",
                "February",
                "Marchuary",
                "Aprilary",
                "Mayary",
        });
        graph.getGridLabelRenderer().setLabelFormatter(staticLabelsFormatter);

        series.setColor(Color.rgb(255, 160, 122));
        series.setDrawDataPoints(true);
        series.setDataPointsRadius(4);
        series.setThickness(4);

        graph.addSeries(series);
        graph.setTitle("BMI changes over the course of last months");
        graph.setTitleTextSize(48);
        graph.setTitleColor(Color.rgb(102, 51, 153));
        graph.getLegendRenderer().setVisible(true);
        graph.getLegendRenderer().setAlign(LegendRenderer.LegendAlign.TOP);

        GridLabelRenderer gridLabel = graph.getGridLabelRenderer();
        gridLabel.setHorizontalAxisTitle("Months");
        gridLabel.setVerticalAxisTitle("BMI");
    }
}

File: BMI_Tipper/app/src/main/java/com/example/tipper/BmiHistoryActivity.kt
package com.example.tipper

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.jjoe64.graphview.GraphView
import com.jjoe64.graphview.LegendRenderer
import com.jjoe64.graphview.helper.StaticLabelsFormatter
import com.jjoe64.graphview.series.DataPoint
import com.jjoe64.graphview.series.LineGraphSeries

class BmiHistoryActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_bmi_history)

        val graph = findViewById<GraphView>(R.id.graph)
        val staticLabelsFormatter = StaticLabelsFormatter(graph)
        val series = LineGraphSeries(arrayOf(
                DataPoint(0.0, 40.0),
                DataPoint(1.0, 30.0),
                DataPoint(2.0, 50.0),
                DataPoint(3.0, 35.0),
                DataPoint(4.0, 45.0)))
        graph.addSeries(series)

        staticLabelsFormatter.setHorizontalLabels(arrayOf(
                "January",
                "February",
                "Marchuary",
                "Aprilary",
                "Mayary"))
        graph.gridLabelRenderer.labelFormatter = staticLabelsFormatter

        series.color = Color.rgb(255, 160, 122)
        series.isDrawDataPoints = true
        series.dataPointsRadius = 4f
        series.thickness = 4

        graph.addSeries(series)
        graph.title = "BMI changes over the course of last months"
        graph.titleTextSize = 48f
        graph.titleColor = Color.rgb(102, 51, 153)
        graph.legendRenderer.isVisible = true
        graph.legendRenderer.align = LegendRenderer.LegendAlign.TOP

        val gridLabel = graph.gridLabelRenderer
        gridLabel.horizontalAxisTitle = "Months"
        gridLabel.verticalAxisTitle = "BMI"
    }
}